compiler/rustc_infer/src/infer/resolve.rs

```diff
@@ -92,7 +92,7 @@ impl<'a, 'tcx> TypeFolder<'tcx> for OpportunisticRegionResolver<'a, 'tcx> {
                     .borrow_mut()
                     .unwrap_region_constraints()
                     .opportunistic_resolve_var(rid);
-                self.tcx().reuse_or_mk_region(r, ty::ReVar(resolved))
+                TypeFolder::tcx(self).reuse_or_mk_region(r, ty::ReVar(resolved))
```

Comment:
Does `FallibleTypeFolder::tcx(self)` not work?

Reply:
Yeah, it does; but what difference?

Reply:
Just thinking about why we can't remove TypeFolder::tcx in favor of just using FallibleTypeFolder::tcx (since it's a super trait)

Reply:
For a `TypeFolder`, the blanket impl of `FallibleTypeFolder::tcx()` just delegates to `TypeFolder::tcx()` — it should get inlined either way for no actual difference in the generated code; and there shouldn't be any semantic difference either.  I only chose `TypeFolder` because that is the trait being implemented, so it was a little less cognitive load (for me anyway).  But I don't mind either way if you think we should call the method on `FallibleTypeFolder`.

Reply:
Because the blanket impl needs to get the tcx from somewhere, but only has a generic `F: TypeFolder`.

Shall I pull the `tcx()` accessor out into its own trait?